test/spacesuit_cors_middleware_test.exs

```diff
@@ -0,0 +1,156 @@
+defmodule SpacesuitCorsMiddlewareTest do
+  use ExUnit.Case
+  doctest Spacesuit.CorsMiddleware
+
+  setup_all do
+    req = %{
+      :headers => %{"origin" => "http://localhost"},
+      :scheme  => "http",
+      :host    => "www.example.com",
+      :port    => 80,
+      :path    => "/matched",
+      :method  => "GET"
+    }
+    {:ok, req: req}
+  end
+
+  describe "handling non-matching request paths" do
```

Comment:
Most of the tests were inspired by [this spec](https://github.com/playframework/playframework/blob/master/framework/src/play-filters-helpers/src/test/scala/play/filters/cors/CORSCommonSpec.scala) and my attempt to express the same here. Due to the inability to change the configuration, as I understand, I wasn't able to cover all the states I would have liked. Perhaps you know of another way to accomplish this.